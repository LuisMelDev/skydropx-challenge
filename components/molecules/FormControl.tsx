import React, { ChangeEventHandler, HTMLInputTypeAttribute } from "react";
import { FormHelperText, Input, Label } from "..";

interface Props {
  labelText: string;
  idInput: string;
  value: any;
  onChange: ChangeEventHandler;
  typeInput: HTMLInputTypeAttribute;
  placeholder?: string;
  textError: string;
  nameInput?: string;
  size?: "full" | "50";
}

export const FormControl = ({
  labelText,
  idInput,
  value,
  onChange,
  typeInput,
  placeholder = "",
  textError = "",
  nameInput = "",
  size = "full",
}: Props) => {
  return (
    <div className={`${size === "full" ? "full block" : "w-5/12 inline-block"} mb-3`}>
      <Label
        htmlFor={idInput}
        text={labelText}
        className="text-sm font-light block"
      />
      <Input
        id={idInput}
        value={value}
        onChange={onChange}
        type={typeInput}
        placeholder={placeholder}
        name={nameInput}
        className="bg-slate-200 outline-none focus:bg-slate-50 focus:shadow font-normal h-10 rounded-lg py-1 px-2 text-md block w-full"
      />
      <FormHelperText text={textError} />
    </div>
  );
};
