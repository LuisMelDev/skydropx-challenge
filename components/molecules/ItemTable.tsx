import React, { MouseEventHandler } from "react";
import { Tab, Td } from "..";

interface ItemTableProps{
    rate: any;
    recomended: boolean;
    onClick: MouseEventHandler<HTMLTableRowElement>
}

export const ItemTable = ({
    rate,
    recomended = false,
    onClick
}: ItemTableProps) => {
    const { provider, service_level_name, days, total_pricing } = rate;
    return (
        <tr className='hover:bg-gray-100 active:bg-gray-200 cursor-pointer transition ease-in-out delay-80' onClick={onClick} >
            <Td>
                <div className='flex items-center'>
                    <p>{provider}</p>
                </div>
            </Td>
            <Td>
                <p>{service_level_name}</p>
            </Td>
            <Td>
                <p>{days} dias</p>
            </Td>
            <Td>
                <p>{total_pricing} MXN</p>
            </Td>
            <Td>{recomended ? <Tab text='Mejor Opcion' /> : null}</Td>
        </tr>
    );
};
