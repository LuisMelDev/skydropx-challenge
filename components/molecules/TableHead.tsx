import React from "react";
import { Th } from "..";

export const TableHead = ({ heads }: { heads: string[] }) => {
    return (
        <thead>
            <tr>
                {heads.map((name) => (
                    <Th name={name} key={name} />
                ))}
            </tr>
        </thead>
    );
};

