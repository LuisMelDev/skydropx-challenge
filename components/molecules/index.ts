export * from "./Nav"
export * from "./FormControl"