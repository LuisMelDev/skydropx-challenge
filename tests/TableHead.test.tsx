import React from "react";
import { shallow } from "enzyme";
import { TableHead } from "../components/molecules/TableHead";

const heads = ["uno", "dos"];

describe("Pruebas <TableHead>", () => {
    let wrapper = shallow(<TableHead heads={heads} />);

    test("debe de mostrarse correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
