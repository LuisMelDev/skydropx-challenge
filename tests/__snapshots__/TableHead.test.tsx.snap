// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <TableHead> debe de mostrarse correctamente 1`] = `
<thead>
  <tr>
    <Th
      key="uno"
      name="uno"
    />
    <Th
      key="dos"
      name="dos"
    />
  </tr>
</thead>
`;
